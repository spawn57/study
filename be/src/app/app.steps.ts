import { defineFeature, loadFeature } from 'jest-cucumber';

const feature = loadFeature(__dirname + `/app.feature`);

defineFeature(feature, (test) => {
  test('Successful login', ({ given, when, then }) => {
    let requestUrl; 
    let result;
    given(/^a user with username "(.*)" and password "(.*)"$/, (username, password) => { 
      requestUrl = '/login';
    });

    when(/the api is triggered/, () => {
      
    });

    then(/it matches the snapshot/, () => {
      expect(true).toMatchSnapshot();
    });
  });
});
