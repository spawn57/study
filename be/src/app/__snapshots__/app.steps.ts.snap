// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Successful login 1`] = `
{
  "header": {
    "connection": "close",
    "content-length": "2",
    "content-type": "text/plain; charset=utf-8",
    "date": "Sat, 06 Jan 2024 07:05:43 GMT",
    "etag": "W/"2-nOO9QiTIwXgNtWtBJezz8kv3SLc"",
    "x-powered-by": "Express",
  },
  "req": {
    "data": {
      "password": "test",
      "username": "test",
    },
    "headers": {
      "content-type": "application/json",
    },
    "method": "POST",
    "url": "http://127.0.0.1:39755/login",
  },
  "status": 200,
  "text": "OK",
}
`;
